### PR TITLE
Add client `RetryingHttpRequesterFilter` example

### PR DESCRIPTION
**Pull request by bondolo**
Motivation:
Users have requested a usage sample for client retry filter
Modifications:
Adds an HTTP example for the client `RetryingHttpRequesterFilter` retry
filter.
Result:
Additional usage example available for users.

### CODE REVIEW

File: servicetalk-examples/http/retry/src/main/java/io/servicetalk/examples/http/retry/RetryClient.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+import io.servicetalk.http.api.HttpClient;
+import io.servicetalk.http.api.HttpResponseStatus;
+import io.servicetalk.http.netty.HttpClients;
+import io.servicetalk.http.netty.RetryingHttpRequesterFilter;
+
+import static io.servicetalk.http.api.HttpSerializers.textSerializerUtf8;
+
+/**
+ * Extends the Async "Hello World" client to immediately retry requests that get a 5XX response. Two attempts will be
```
Comment (tkountis) [inline servicetalk-examples/http/retry/src/main/java/io/servicetalk/examples/http/retry/RetryClient.java:26]: "Two attempts" --> "up-to two attempts"

File: servicetalk-examples/http/retry/src/main/java/io/servicetalk/examples/http/retry/RetryClient.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+ * made before failure.
+ */
+public final class RetryClient {
+    public static void main(String[] args) throws Exception {
+        try (HttpClient client = HttpClients.forSingleAddress("localhost", 8080)
+                .appendClientFilter(new RetryingHttpRequesterFilter.Builder()
+                        .responseMapper((httpResponseMetaData -> {
+                            // If the response status is 500-599 then request a retry
+                            HttpResponseStatus.StatusClass status = httpResponseMetaData.status().statusClass();
+                            return HttpResponseStatus.StatusClass.SERVER_ERROR_5XX == status ?
```
Comment (tkountis) [inline servicetalk-examples/http/retry/src/main/java/io/servicetalk/examples/http/retry/RetryClient.java:36]: consider `SERVER_ERROR_5XX.contains(meta.status())`
also consider static imports in this area, all these inner classes create unnecessarily code noise.

File: servicetalk-examples/http/retry/src/main/java/io/servicetalk/examples/http/retry/RetryClient.java (hunk 1 above)
Comment (bondolo) [reply servicetalk-examples/http/retry/src/main/java/io/servicetalk/examples/http/retry/RetryClient.java:26]: I was definitely unsure whether there would be two attempts or two retries, additional attempts following an initial failure. I will correct this usage but I did find it ambiguous in the API docs as to the total number of attempts that would be made, often due to whether the first attempt counted as a retry or not.

**Review by tkountis — APPROVED** [review]
minor comments then lgtm

File: servicetalk-examples/http/retry/src/main/java/io/servicetalk/examples/http/retry/RetryClient.java (hunk 1 above)
Comment (tkountis) [reply servicetalk-examples/http/retry/src/main/java/io/servicetalk/examples/http/retry/RetryClient.java:26]: good point, its up-to two retries, excluding the initial request.

File: servicetalk-examples/http/retry/src/main/java/io/servicetalk/examples/http/retry/RetryClient.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                                    new RetryingHttpRequesterFilter.HttpResponseException(
+                                            "Retry 5XX", httpResponseMetaData)
+                                    // Not a 5XX response, someone else can deal with it.
```
[the remark this answers is not in the document]
Comment (bondolo) [reply servicetalk-examples/http/retry/src/main/java/io/servicetalk/examples/http/retry/RetryClient.java:39]: My version of the comment is probably too informal, but the mapper is deciding less than "no need to retry" as other facilities could decide that retry is needed. It is an ambiguity of "do not retry" vs "I was unable to determine if retry is needed."

**Pull request merged**